packages/core/RNSentryAndroidTester/app/src/test/java/io/sentry/rnsentryandroidtester/RNSentryCompositeOptionsConfigurationTest.kt

```diff
@@ -0,0 +1,51 @@
+package io.sentry.rnsentryandroidtester
+
+import io.sentry.Sentry.OptionsConfiguration
+import io.sentry.android.core.SentryAndroidOptions
+import io.sentry.react.RNSentryCompositeOptionsConfiguration
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+
+@RunWith(JUnit4::class)
+class RNSentryCompositeOptionsConfigurationTest {
+    @Test
+    fun `configure should call base and overriding configurations`() {
+        val baseConfig: OptionsConfiguration<SentryAndroidOptions> = mock()
+        val overridingConfig: OptionsConfiguration<SentryAndroidOptions> = mock()
+
+        val compositeConfig = RNSentryCompositeOptionsConfiguration(baseConfig, overridingConfig)
+        val options = SentryAndroidOptions()
+        compositeConfig.configure(options)
+
+        verify(baseConfig).configure(options)
+        verify(overridingConfig).configure(options)
+    }
+
+    @Test
+    fun `configure should apply base configuration and override values`() {
+        val baseConfig =
+            OptionsConfiguration<SentryAndroidOptions> { options ->
+                options.dsn = "https://base-dsn@sentry.io"
+                options.isDebug = false
+                options.release = "some-release"
```

Comment:
I think `options.environment = "base"` is missing here.

Reply:
I took a look at it again and I now understand the different scenarios, it's good as is.